Modules/posixmodule.c

```diff
@@ -5302,7 +5302,7 @@ os_spawnve_impl(PyObject *module, int mode, path_t *path, PyObject *argv,
         PyMem_DEL(envlist[envc]);
     PyMem_DEL(envlist);
   fail_1:
-    free_string_array(argvlist, lastarg);
+    free_string_array(argvlist, lastarg + 1);
```

Comment:
`lastarg` is set to `i` on line 5265 when `fsconvert_strdup` fails. Using `lastarg + 1` in this case will free one too many pointers. It's also freeing one too many when all of `argv` is converted successfully and `lastarg` is set to `argc`. In this case `argvlist[argc]` is a `NULL` pointer, so freeing it is harmless.

I think the original code only needs a simple, single-line change. For the empty string case on line 5269. `fsconvert_strdup` has succeeded, so we know there's exactly `lastarg = 1` pointer to free.